src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -488,16 +505,31 @@ async function loadTimezone(timeZoneResource: LoadingResource) : Promise<void> {
   removeRunDependency(runDependencyId);
 }
 
-async function loadICUData(icuDataResource: LoadingResource) : Promise<void> {
+function getICUResourceName(bootConfig: BootJsonData, culture: string | undefined): string {
+  const combinedICUResourceName = 'icudt.dat';
+  if (!culture || bootConfig.icuDataMode == ICUDataMode.All) {
+    return combinedICUResourceName;
+  }
+
+  const prefix = culture.split('-')[0];
```

Comment:
@lewing

Reply:
Yeah, it would be really good if this logic was inside the runtime, for example as a function like `MONO.getShardFilenameForCulture(cultureName)`.

We don't want to have to coordinate changes to this across runtime and framework if the sharding set changes over time.

@lewing, do you think that's possible in time for RC2?

Reply:
it should be easy to add, I thought you might want to rename the files or add more variations of icudt (or user might want to do it) since it's not difficult to create an icudt for any culture(s) user wants

Reply:
Thanks @EgorBo, that would be great! Yes it would be good if the runtime had a function to provide info about the default built-in sharding setup. Theoretically we could still vary the behavior at the application layer, but we have no plans to vary it right now.

Do you want me to file an issue on the runtime repo to represent this?

Reply:
@pranavkm Presumably we could merge this PR as-is and then switch to the runtime logic later, just to avoid being blocked.

Reply:
@SteveSandersonMS I'll file a PR shortly with a prototype.

Reply:
I see it's here: https://github.com/dotnet/runtime/pull/41795

Thanks @EgorBo!

Reply:
@SteveSandersonMS so should I backport that API to 5.0rc2 ?